src/foam/core/stdlib.js

```diff
@@ -746,7 +759,24 @@ foam.LIB({
     methods: [
       function clone(o)      { return typeOf(o).clone(o); },
       function equals(a, b)  { return typeOf(a).equals(a, b); },
-      function compare(a, b) { return typeOf(a).compare(a, b); },
+      function compare(a, b) {
+        // To ensure that symmetry is present when comparing,
+        // we will always use the comparator of higher precedence.
+        var types = foam.typeOrder;
+        var typeID = function(obj) {
```

Comment:
No, that's not what I mean. I first compare the types and if they're different, I don't need to compare the actual values. If they are the same type the first compare returns 0 so I then do the type-specific compare().